include/hal/library/secretlib/secretlib.h

```diff
@@ -0,0 +1,75 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2021-2023 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+#ifndef SECRETLIB_H
+#define SECRETLIB_H
+
+#include "internal/libspdm_lib_config.h"
+#include "hal/base.h"
+#include "industry_standard/spdm.h"
+#include "library/spdm_return_status.h"
+
+#if LIBSPDM_ENABLE_CAPABILITY_PSK_EX_CAP
+/**
+ * Derive HMAC-based Expand key Derivation Function (HKDF) Expand, based upon the negotiated HKDF
+ * algorithm.
+ *
+ * @param  base_hash_algo  Indicates the hash algorithm.
+ * @param  psk_hint        Pointer to the peer-provided PSK Hint.
+ * @param  psk_hint_size   PSK Hint size in bytes.
+ * @param  info            Pointer to the application specific info.
+ * @param  info_size       Info size in bytes.
+ * @param  out             Pointer to buffer to receive HKDF value.
+ * @param  out_size        Size of HKDF bytes to generate.
+ *
+ * @retval true   HKDF generated successfully.
+ * @retval false  HKDF generation failed.
+ **/
+extern bool libspdm_psk_handshake_secret_hkdf_expand(
```

Comment:
I am thinking if we really need this standalone secret lib.

Can we duplicate that function in both requester lib and responder lib, since they are only 2 PSK optional functions?

Reply:
I will try that out.